config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -200,6 +204,8 @@ private static Object getListValue(Class<?> beanClass, Type parameterType, Class
                 beanList.add(createInternal(listMember, (Class<?>) elementType));
             }
             return beanList;
+        } else if (((Class<?>) elementType).isEnum()) {
+            return config.getEnumList((Class<Enum>) elementType, configPropName);
```

Comment:
did you consider pros and cons of putting this in front of hasAtLeastOneBeanProperty?
couldn't an enum class have a bean property?